readium/shared/src/main/java/org/readium/r2/shared/util/downloads/DownloadManager.kt

```diff
@@ -85,11 +92,11 @@ public interface DownloadManager {
 
     /**
      * Submits a new request to this [DownloadManager]. The given [listener] will automatically be
-     * registered.
+     * registered. Requests already submitted will be ignored.
      *
      * Returns the ID of the download request, which can be used to cancel it.
      */
-    public fun submit(request: Request, listener: Listener): RequestId
+    public fun submit(request: Request, listener: Listener)
```

Comment:
It could be worth mentioning that we *must* call `remove()` when we're done in the `Listener`, no matter the result.